mm0-rs/src/elab/frozen.rs

```diff
@@ -122,15 +122,15 @@ impl FrozenAtomData {
 
 /// A wrapper around a [`LispVal`](../lisp/struct.LispVal.html) that is frozen.
 #[derive(Debug)]
-pub struct FrozenLispVal(LispVal);
+pub struct FrozenLispVal(pub LispVal);
```

Comment:
These can't be public, it's part of the safety invariant. You should use the (unsafe) `thaw()` function instead to get a handle on the underlying lisp value. The safety property you have to ensure is that you don't call these functions after the elaborator is finished, meaning inside server code. Currently I am not strictly adhering to this because a lot of printing functions haven't been converted yet, and I haven't tripped over any race conditions yet, but you should beware of races on the `Rc` reference count.